In this exercise, you will practice basic `dplyr` (the relevant package within `tidyverse`) operations on a dataset of gene expression levels across human tissues (from the GTEx Project). The dataset (`dt`) has one row per gene, with columns:

-   `Name`: the Ensembl gene ID\
-   `Description`: the gene symbol
-   One column per tissue containing the median expression levels (in TPM units)

------------------------------------------------------------------------

### Start by loading the data into R:

Note the arguments, which specify that the first two lines of the file will be skipped, that the file contains a header line that should be used to form column names, and that the columns are tab (`\t`) separated.

```{r}
setwd('/Users/cmdb/Data/GTEx')
dt <- read_delim("GTEx_Analysis_2017-06-05_v8_RNASeQCv1.1.9_gene_median_tpm.gct.gz", skip = 2, col_names = TRUE, delim = "\t")
spec(dt)
row_names_vector <- rownames(dt)
print(row_names_vector)
```

### Q1. How many genes are in the dataset, and how many tissues are represented?

```{r}

# code and answer here
# column_Liver_vector <- data.frame('Liver')
# print(column_Liver_vector)
n_genes <- nrow(dt)
n_tissues <- ncol(dt) - 2
n_genes
n_tissues
# There are 56200 genes
# There are 54 tissues
```

------------------------------------------------------------------------

### Q2. How many genes are expressed above a given threshold in each tissue?

A common threshold is 1 TPM to define whether a gene is "expressed."\
Compute, for each of the three tissues, `Liver`, `Testis`, and `Whole Blood`, how many genes are expressed above this threshold.\
Then identify which tissue has the **most** and which has the **fewest** expressed genes. See this paper for one possible explanation: Xia et al. 2021: [PMC7891839](https://pmc.ncbi.nlm.nih.gov/articles/PMC7891839/).

Do the results change if you set the threshold higher, say at 10 TPM?

```{r}

# Code here
dt %>%
  select(Liver) %>%
  filter(Liver > 1)
# Liver: 13,499
dt %>%
  select(Testis) %>%
  filter(Testis > 1)
# Testis: 25,064
dt %>%
  select(`Whole Blood`) %>%
  filter(`Whole Blood` > 1)
# Whole Blood: 11,616
dt %>%
  select(Liver) %>%
  filter(Liver > 10)
dt %>%
  select(Testis) %>%
  filter(Testis > 10)
dt %>%
  select(`Whole Blood`) %>%
  filter(`Whole Blood` > 10)
# Interpretation here as a comment
# The testis has the most expressed genes.
# The results do change if the TPM is changed to 10.
dt

```

------------------------------------------------------------------------

### Q3. What are the 20 most highly expressed genes in the liver?

What do you notice about many of the genes in this list? Can you propose a hypothesis to explain your observation?

```{r}

# Code here
top20_liver <- dt %>%
  arrange(desc(Liver)) %>%
  slice(1:20) %>%
  select(Description, Liver)

top20_liver
# Interpretation here as a comment
# Most of these genes encode mitochondrial proteins. This makes sense given that the liver is primarily a metabolic organ. One of the principal functions of the liver is to take glucose and generate ATP from it, which mostly occurs in the mitochondria. Of note is the high level of expression of the ALB gene. The ALB gene encodes the protein albumin. While albumin is not exlcusively involved in metabolism, it stands to reason that its expression would be elevated in a very metabolically active organ like the liver, given its role as an antioxidant -- molecules crucial for eliminating the ROS formed during cellular respiration.
```

------------------------------------------------------------------------

### Q4. Compare the mean expression levels of mitochondrial versus nuclear-encoded genes in the Liver.

Genes encoded by the mitochondrial genome have the gene symbol prefix `MT-`. The function `str_starts()` from the `stringr` package in `tidyverse` tests whether a string (or a vector of strings) start with a given pattern, returning TRUE/FALSE (or a boolean vector) as output.

Use `str_starts()` to identify which genes are mitochondrially encoded and store this information in a new Boolean column named `is_mitochondrial`. Next, compute the mean and median expression of mitochondrial genes in the liver, as well as the mean and median expression of nuclear-encoded genes in the liver and compare them. Do mitochondrial genes exhibit higher or lower mean and median expression than nuclear genes? What does the difference between the mean and median tell you about the distribution of gene expression.

```{r}
# Code here

dt <- dt %>%
  mutate(is_mitochondrial = str_starts(Description, "MT-"))

table(dt$is_mitochondrial)

mito_stats <- dt %>%
  filter(is_mitochondrial) %>%
  summarise(
    mean_expr = mean(Liver),
    median_expr = median(Liver)
  )

nuclear_stats <- dt %>%
  filter(!is_mitochondrial) %>%
  summarise(
    mean_expr = mean(Liver),
    median_expr = median(Liver)
  )

mito_stats
nuclear_stats

dt

# Interpretation here as a comment
# The mitochondrial genes exhibit higher mean and median expression values in the liver than nuclear genes.
# The mean expression value for mitochondrial genes is 11,592.72, while the median expression value is 3.40258. This suggests that the expression of mitochondrial genes in the liver is positively skewed. That is, there are some mito genes that are very highly expressed (i.e., have very high expression values), but most of the genes are actually expressed at low levels. The mean expression value for nuclear genes in the liver is 8.373839 and the median is 0. This suggests that nuclear-encoded genes in general are expressed at lower levels than mito-encoded genes in the liver, with many of them not being expressed at all.
```

------------------------------------------------------------------------

### Q5. Are liverâ€™s most highly expressed genes also highly expressed in another tissue?

Take the 20 most highly exprssed genes in **Liver** and check how they are expressed in the **Heart - Left Ventricle**. Identify:\
1. one gene that is highly expressed in both tissues\
2. one gene that is highly expressed in Liver but very low in Heart

Look up basic information about these genes on the internet. Do their expression patterns make sense given their known functions?

```{r}

# Code here
top20_liver <- dt %>%
  arrange(desc(Liver)) %>%
  slice(1:20)

top20_liver %>%
  select(Description, Liver, `Heart - Left Ventricle`)

top20_liver

# Interpretation here as a comment
# MT-ND4 is highly expressed in both the liver and heart LV, with expression values of 58,775.30 and 66,015.7, respectively. This makes sense beacuse all tissues require expression of mito genes as all tissues require the proteins necessary for ATP production. Albumin is highly expressed in the liver but not in the heart LV, which makes sense since the liver secretes albumin.
```

------------------------------------------------------------------------

### Q6. Comparing gene expression between tissues

Another way to compare tissues is to ask how similar are their expression patterns overall, across all genes?

R has a built-in function called `cor.test()` that measures the correlation between two vectors of numbers. Remember that you can extract the contents of a given column of a tibble as a vector using the syntax `tibble_name$column_name`. If a column name contains spaces, enclose the column name in backticks such as `` tibble_name$`column name` ``.

Use `cor.test()` with the argument `method = "kendall"` to compute: - the Spearman correlation between `Liver` and `Heart - Left Ventricle` - the Spearman correlation between `Liver` and `Brian - Cortex`

Which pair of tissues is more strongly correlated? Does this result make sense biologically?

```{r}
# Liver vs Heart answer
cor.test(dt$Liver, dt$`Heart - Left Ventricle`, method = "kendall")

# Liver vs Brain answer
cor.test(dt$Liver, dt$`Brain - Cortex`, method = "kendall")

# interpretation

# Kendall's rank correlation coefficient is greater for the liver and heart LV than for the liver and brain cortex. The former is calculated to be 0.8053218 and the latter 0.7342544. This does make sense as the cells of the brain cortex are more heterogeneous than those of the liver and heart. The cortex of the brain is made up of many different types of neural cells with distinct transcriptional profiles, whereas the liver is largely made up of hepatocytes and the heart is mostly made up of cardiac muscle cells. This diversity of expression in the tissue of the cortex could potentially be introducing variation in expression values, thereby decreasing the correlation between the expression of the same genes in the liver.
```

------------------------------------------------------------------------

### Optional: further exploration

What other patterns are present in the data? Use tidyverse functions to explore and summarize the data in different ways and see what you can learn.

```{r}
# answer here
```
